proposals/jib-cli-surface.md

```diff
@@ -0,0 +1,71 @@
+# Jib CLI Surface Specification
+
+Specification for a command structures exposed to users of the jib-cli
+
+## Usage
+`jib COMMAND [OPTIONS]`
+
+## Commands
+
+The jib cli exposes the following top level commands
+```
+build        build a container
+```
+
+## Options
+
+### Required
+```
+-t  --target <image-ref>  a target image reference, using the url schema defined in jib
+                          gcr.io/project/image (default is build to registry)
+                          registry://gcr.io/project/image
+                          docker://some-image-ref
+                          tar://relative/path.tar
+                          tar:///aboslute/path.tar
+```
+
+### Optional
+
+
+#### Build Config
+```
+-b, --build-file <file>                    location of the build file (default <context>/jib.yaml)
+-c, --context <dir>                        location of the build context (default .)
+    --docker-config <directory>            location of docker configuration
+    --name <image-ref>                     image name to bake into tar file (only available when using -t tar://...)
+-p, --parameter <name>=<value>             templating parameters replace `${name}` with `value` in the build file (repeatable)
+    --tag <tag1>[,<tag2>,...]              additional tags for target
+```
+
+#### Auth/Security
+```
+    --allow-insecure-registries            allow jib to communicate with registries over https
+    --credHelper <registry>=<credHelper>   credential helper to use for a registry (repeatable)
+    --send-credentials-over-http           allow jib to send credentials over http (used in conjunction with --allow-insecure-registries)
+    --to-auth <username> <password>        configure a username and password for authentication on the registry that an image is being built to
```

Comment:
I think maybe I might split this into
```
--to-auth-username <username>
--to-auth-password <password>
```
this lets us take advantage of some password related tools that pico cli adds